components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/service/BlackListWhiteListService.java

```diff
@@ -56,68 +59,62 @@ public BlackListWhiteListService()
 	 * sign if not all the process fails. The entire process run as a single
 	 * transaction .
 	 */
-	public void blacklist(BlackListDTO dto) throws BusinessException {
+	public APIBlacklistWhitelistResponseDTO blacklist(BlacklistWhitelistDTO dto) throws BusinessException {
 
-		String[] msisdns = dto.getUserMSISDN();
-		MSISDNValidationDTO msisdnValidationDTO = new MSISDNValidationDTO();
-
-		final String apiID_ = dto.getApiID();
-		final String apiName_ = dto.getApiName();
-		final String userId_ = dto.getUserID();
-
-
-			try{
-
-				msisdnValidationDTO.setValid(Arrays.asList(msisdns));
-				msisdnValidationDTO.setValidationRegex(dto.getValidationRegex());
-				msisdnValidationDTO.setValidationPrefixGroup(dto.getValidationPrefixGroup());
-				msisdnValidationDTO.setValidationDigitsGroup(dto.getValidationDigitsGroup());
-				msisdnValidationDTO.process();
+		List<String> valid = new ArrayList<>();
+		List<String> invalid = new ArrayList<>();
+		APIBlacklistWhitelistResponseDTO responseDTO = new APIBlacklistWhitelistResponseDTO(0,0);
+		int success = 0;
 
+		Matcher matcher;
+		try {
 
-				// load already black listed numbers
-				List<MsisdnDTO> alreadyBlacklisted;
-
-				alreadyBlacklisted = dao.getBlacklisted(apiID_);
-
-				// Remove already black listed from the list
-				for (MsisdnDTO msisdn : alreadyBlacklisted) {
-					msisdnValidationDTO.getValidProcessed().remove(msisdn);
+			for (String m : dto.getMsisdnList()) {
+				matcher = bwRegex.matcher(m);
```

Comment:
It will be further clearer if you move this msisdn validation into a common validator service class since this is one of the core validations done within the service.

Reply:
it was decided to keep all the business logic in the service layer